src/poetry/packages/locker.py

```diff
@@ -244,7 +244,7 @@ def set_lock_data(self, root: Package, packages: list[Package]) -> bool:
 
         if root.extras:
             lock["extras"] = {
-                extra: [dep.pretty_name for dep in deps]
+                extra: [dep.pretty_name for dep in sorted(deps, key=lambda dep: dep.pretty_name)]
```

Comment:
quibble: this feels like a complicated way of writing `sorted(dep.pretty_name for dep in deps)`

Reply:
Thanks for the feedback. I changed to your suggestion (and also fixed the flake error)